dipy/reconst/tests/test_ivim.py

```diff
@@ -0,0 +1,56 @@
+""" Testing DTI
+
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from nose.tools import (assert_true, assert_equal,
+                        assert_almost_equal, assert_raises)
+import numpy.testing as npt
+from numpy.testing import (assert_array_equal, assert_array_almost_equal,
+                           assert_)
+import nibabel as nib
+
+import scipy.optimize as opt
+
+from dipy.reconst import ivim as ivim
+from dipy.reconst.ivim import ivim_function
+from dipy.data import get_data, dsi_voxels, get_sphere
+import dipy.core.gradients as grad
+from dipy.sims.voxel import single_tensor
+from dipy.io.gradients import read_bvals_bvecs
+from dipy.core.gradients import (gradient_table, GradientTable,
+                                 gradient_table_from_bvals_bvecs,
+                                 reorient_bvecs)
+from dipy.sims.voxel import multi_tensor
+
+
+def test_nlls_fit():
+    """
+    Test the implementation of NLLS
+    """
+    fimg, fbvals, fbvecs = get_data('small_101D')
+    bvals, bvecs = read_bvals_bvecs(fbvals, fbvecs)
+    gtab = gradient_table(bvals, bvecs)
+
+    f, D_star, D = 90, 0.0015, 0.0008
+
+    mevals = np.array(([D_star, D_star, D_star], [D, D, D]))
+    # This gives an isotropic signal
+
+    signal = multi_tensor(gtab, mevals, snr=None, fractions=[f, 100 - f])
```

Comment:
Fixes to make things more consistent would be most welcome!

On Thu, May 26, 2016 at 9:18 AM, Eric Peterson <notifications@github.com>
wrote:

> In dipy/reconst/tests/test_ivim.py
> <https://github.com/nipy/dipy/pull/1058#discussion_r64775504>:
>
> > +
> > +
> > +def test_nlls_fit():
> > +    """
> > +    Test the implementation of NLLS
> > +    """
> > +    fimg, fbvals, fbvecs = get_data('small_101D')
> > +    bvals, bvecs = read_bvals_bvecs(fbvals, fbvecs)
> > +    gtab = gradient_table(bvals, bvecs)
> > +
> > +    f, D_star, D = 90, 0.0015, 0.0008
> > +
> > +    mevals = np.array(([D_star, D_star, D_star], [D, D, D]))
> > +    # This gives an isotropic signal
> > +
> > +    signal = multi_tensor(gtab, mevals, snr=None, fractions=[f, 100 - f])
>
> Looks like dipy may be a little conflicted also. Single_tensor uses S0=1
> and multi_tensor uses S0=100, though others seem to prefer the 100 area.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/1058/files/951c19fe208c5d54250cf7088c16b962fad29d51#r64775504>
>